synapse/python_dependencies.py

```diff
@@ -101,7 +101,7 @@
     "saml2": [
         "pysaml2>=4.5.0",
     ],
-    "oidc": ["authlib>=0.14.0"],
+    "oidc": ["authlib>=0.14.0", "pyjwt>=1.6.4"],
```

Comment:
We annoyingly list `pyjwt` twice now, which seems like it has potential for things to go haywire if we don't keep the versions in sync. I'm unsure what to do about that? (Does `authlib` depend on `pyjwt` by the way, or are they completely separate?)

Reply:
They are completely separate unfortunately :(

Reply:
I'm not sure this is a problem — do we have reason to believe that pip won't choose the most restrictive requirement given that it should already be able to cope if a transitive requirement also diverges on what version it needs?

Alternatively if this really does concern us, it could be replaced with a `PYJWT = "pyjwt>=1.6.4"` constant and used in both places, since it's just code after all :).

Reply:
It should be fine unless they conflict, I agree! 👍

I was thinking that we probably would want to keep them the same, but reconsidering; I'm unsure if that's true. If our support for JWT logins requires a newer version than we might want to bump one and not the other. This might be confusing for packagers of course. 😄